### PR TITLE
feat: Partial support for Qute Type-safe Message Bundles (Java classes only)

### LINKED ISSUE #1139: Partial support for Qute Type-safe Message Bundles (Java classes only)
See PR https://github.com/redhat-developer/quarkus-ls/pull/845 from Quarkus LS

Here a  demo of this qute message support in IJ:

QuteMessageDemoInIJ

This issue provides basic support for https://quarkus.io/guides/qute-reference#type-safe-message-bundles where message value are deifined in Java files in the @Message annotation.

```java
import io.quarkus.qute.i18n.Message;
import io.quarkus.qute.i18n.MessageBundle;

@MessageBundle 
public interface AppMessages {

    @Message("Hello {name}!") 
    String hello_name(String name); 
}
```

but doesn't support where message value is defined in a msg.properties

```java
import io.quarkus.qute.i18n.Message;
import io.quarkus.qute.i18n.MessageBundle;

@MessageBundle 
public interface AppMessages {

    @Message
    String hello_name(String name); 
}
```

`msg.properties:`

`hello_name=Hello {name}!`

### PR DESCRIPTION
**Pull request by angelozerr**
feat: Partial support for Qute Type-safe Message Bundles (Java classes only)

Fixes #1139

### CODE REVIEW

File: src/main/java/com/redhat/devtools/intellij/qute/psi/internal/template/datamodel/TypeSafeMessageBundlesSupport.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0git che
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/qute/psi/internal/template/datamodel/TypeSafeMessageBundlesSupport.java:7]: something wrong happened to that header

Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/qute/psi/internal/template/datamodel/TypeSafeMessageBundlesSupport.java]: fixed

File: src/test/java/com/redhat/devtools/intellij/qute/psi/template/TemplateGetDataModelProjectForMessagesTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		// String hello();
+		ValueResolverInfo message = assertValueResolver("msg2", "hello() : java.lang.String", "org.acme.App2Messages",
+				null, false, resolvers);
+		assertMessageResolverData(null, "HELLO!", message);
+
+		// @Message("HELLO!")
+		// String hello();
+		message = assertValueResolver("msg2", "hello2() : java.lang.String", "org.acme.App2Messages", null, false,
+				resolvers);
+		assertMessageResolverData(null, null, message);
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply src/test/java/com/redhat/devtools/intellij/qute/psi/template/TemplateGetDataModelProjectForMessagesTest.java:107]: It was the comment which was wrong, fixed.

**Comment by fbricon** [comment]
@angelozerr thanks!

**Pull request merged**